core/src/main/java/io/kestra/core/runners/ExecutorService.java

```diff
@@ -559,7 +565,12 @@ private Executor handleKilling(Executor executor) {
         );
 
         if (executor.getExecution().hasRunning(currentTasks) || executor.getExecution().hasCreated()) {
-            return executor;
+            // if some tasks are not killed, send a killing event to the worker
+            killQueue.emit(ExecutionKilled
```

Comment:
One things that annoy me is that you break the transaction here, you emit directly on the queue instead of returning to executor object, that will insert into with the same transaction from kafka or jdbc.

Reply:
For JDBC everything is done inside the transaction when consuming from the execution queue, for Kafka I'm not sure.

One way to do it would be to add an `ExecutionKilled` property inside the Executor object and send the ExecutionKilled message from the Jdbc/Memory/KAfka executor instead.

Reply:
This is the goal exactly. 
The `killQueue.emit` will be outside of the transaction also on the jdbc (since we don't provide the `DSLContext`). 
(but there is a subject to have transaction globally on jdbc executor that we don't have for now)

Reply:
I refacto to send ExecutionKilled from the executor and not the executor service.